tests/transforms/preprocessing/test_resample.py

```diff
@@ -27,6 +28,19 @@ def test_reference(self):
                 ref_image_dict[DATA].shape, image_dict[DATA].shape)
             assert_array_equal(ref_image_dict[AFFINE], image_dict[AFFINE])
 
+    def test_coregistration(self):
+        spacing = 1
+        coregistration_name = 'coregistration'
+        transform = Resample(spacing, coregistration=coregistration_name)
+        transformed = transform(self.sample)
+        for image_dict in transformed.values():
+            if coregistration_name in image_dict.keys():
+                new_affine = np.eye(4)
+                new_affine[0, 3] = 10
+                assert_array_equal(image_dict[AFFINE], new_affine)
+            else:
+                assert_array_equal(image_dict[AFFINE], np.eye(4))
```

Comment:
The transform does not seem to be working, and I think this test is not useful.

The images never contain `coregistration_name`, that goes into the sample dict, so the test jumps directly to line 42, which is True just because the test sample has an identity affine by default.

Reply:
This is the script I used to try the transform:

```python
import numpy as np
from torchio.transforms import Resample
from torchio import Image, ImagesDataset, INTENSITY, Subject

matrix = np.fromstring(
    """1.03324 -0.00579961 -0.0387759 -4.59106
0.00273174 0.939307 0.297793 -5.97419
0.0310681 -0.316461 1.02389 -0.777692
0 0 0 1
""", sep=' ')

subject = Subject(
    t1=Image('~/Dropbox/MRI/t1.nii.gz', INTENSITY),
    mni=Image('~/episurg/mri/mni_icbm152_nl_VI_nifti/icbm_avg_152_t1_tal_nlin_symmetric_VI.nii.gz', INTENSITY),
    to_mni=matrix,
)
subjects_list = [subject]
transform = Resample('mni', coregistration='to_mni')
dataset = ImagesDataset(subjects_list, transform=transform)
transformed = dataset[0]
dataset.save_sample(transformed, dict(t1='/tmp/t1_on_mni.nii.gz'))
```

Reply:
Ok, I will investigate but I can see the effect of the coregistration on the samples I'm looking at, the first is resampled using coregistration while the other one is resampled without it:
![with_coregistration](https://user-images.githubusercontent.com/39873986/80111402-2453c800-8580-11ea-8c75-9d53227399b1.png)
![without_coregistration](https://user-images.githubusercontent.com/39873986/80111403-2584f500-8580-11ea-81cb-58258b13df2a.png)

Reply:
And I have runned the test with a print inside to check and it goes inside the if statement. As I have changed `self.sample`, the t1 image has a coregistration attribute but not the t2.

Reply:
Oh yes, I got the error with your snippet, the coregistration key (in your example "to_mni") must be in the image, not in the subject. It is done this way because it is expected that if a subject has several images they could all have their own coregistration.

Reply:
Oh, you pass the matrix to each image. I thought it was per sample, as you can see in my example. The transform should issue a warning or error if this happens (the argument is not `None`, but the key is not found in any image in the sample). Can you please add that check?

Reply:
Yes you right, I add this check to prevent further problems!